doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,150 @@
+generate reports in a performant manner. We currently use heavy layers of caching to mitigate these performance
+concerns, but the less caching we _need_ to do, the better.
+
+### Priorities and constraints
+
+The upcoming changes to our reporting pipeline are quite new to us, so the priority is the ability to iterate.
+Performance is a secondary concern, since we are not yet at a scale where we need to worry about it too much, and our
+caching strategy works well today.
+
+## Decision
+
+We will introduce a set of reporting tables, implemented as materialized database views, to start powering our reporting
+requirements. This reporting schema will transform the data in our application tables into a more reports-friendly
+format, exposing convenient building blocks to build reports with straightforward SQL queries.
+
+Once the reporting tables are populated, we will use Metabase to create and maintain the SQL queries that run against
+the reporting tables to generate reports.
+
+```
+ --------------------             ------------------------------------             ------------------
+| Application tables |    -->    | Reporting tables as database views |    -->    | Metabase reports |
+ --------------------             ------------------------------------             ------------------
+eg. blood_pressures              eg. blood_pressures_over_time                    eg. monthly control rate query
+                                 (latest BP for each patient as of every month)   (as of every month, how many BPs were
+                                                                                   controlled and recent)
+```
+
+The reporting tables will be implemented as database views within the Simple database.
+
+### Reporting Tables
+
+To start, the reporting tables will be granular de-normalized records of patients, blood pressures, encounters, and
+prescribed drugs.
+
+* Time: Each record will feature a month as dimension of time
+* Location: Each record will feature a facility (facility ID) as dimension of location
+
+For example, if "Ashish" is a patient in Simple, the "blood pressures over time" table will have records like:
+* As of Jan 2020, what was Ashish's most recent BP?
+* As of Feb 2020, what was Ashish's most recent BP?
+* As of Mar 2020, what was Ashish's most recent BP?
+* ...and so on
+
+The "prescription drugs over time" table will have records like:
+* As of Jan 2020, what drugs were prescribed to Ashish?
+* As of Feb 2020, what drugs were prescribed to Ashish?
+* As of Mar 2020, what drugs were prescribed to Ashish?
+* ...and so on
+
+Since these tables are pre-computing questions like "What was the most recent BP as of X month", the reports queries
+become much simpler. Instead of dealing with this question, the reports query can instead fetch all records for a
+particular month, and the "most recent" information will be readily available in the record.
+
```

Comment:
I think it might be useful to link to a question being answered via the reporting schema here. Perhaps to the other open PR, and a metabase question?